content/numpy/concepts/math-methods/terms/arcsin/arcsin.md

```diff
@@ -0,0 +1,72 @@
+---
+Title: '.arcsin()'
+Description: 'Calculate the inverse sine of each element in an array.'
+Subjects: 
+  - 'Computer Science'
+Tags: 
+  - 'Methods'
+  - 'Trigonometry'
+  - 'NumPy'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+  - 'paths/data-science'
+  - 'paths/data-science-foundations'
+---
+
+The **'.arcsin()'** method is used to calculate the inverse sine of the elements in an array. It is commonly employed when solving trigonometry problems where an angle of a right sided triangle is unknow, but two of the sides are known. The result should always be in the range -pi/2 to pi/2 (radians).
+
+Conversion of radians to degrees is done by the formula $r*180/\pi$.
+
+## Syntax
+
+```pseudo
+numpy.arcsin(x, out=None, where=True)
+```
+
+Parameters:
+
+- `x`: array
+  - y-coordinate on the unit circle which makes input in the range from -1 to 1.
+  - If input is outside range, then nan is returned as a result.
+- `out` (Optional):
+  - Array for result. If not provided or None, a freshly allocated array is returned.
+- `where` (Optional): array_like
+  - The condition (array of boolean values) that determines the elements on which the method is to be applied.
+  - Output that corresponds to `False` for a particular element will be returned as is.
+  - If not provided, the arcsin is calculated for all elements in the input array.
+
+Returns: radian: array
+
+## Example
+
+The below example shows the `.arcsin()` method in use:
+
+```py
+# Importing the 'numpy' library as 'np'
+import numpy as np
+
+# Defined input array
+a_input = [1, 0.5, 0, -0.5, -1]
+
+# Computing the arcsin of all elements in an array
+result_all = np.arcsin(a_input)
+print("result_all:", result_all)
+
+# Computing the arcsin of specific elements in the array
+result_spec = np.arcsin(a_input, where=np.array([True, True, False, True, False]))
+print("result_spec:", result_spec)
+
+# Computing the arcsin of all elements greater than 0
+result_gt0 = np.arcsin(a_input, where=list(map(lambda x: x>0, a_input)))
+print("result_gt0:", result_gt0)
+
+```
+
+The output of the above code is shown below:
+
+```shell
+result_all:  [ 1.57079633  0.52359878  0.         -0.52359878 -1.57079633]
+result_spec: [ 1.57079633  0.52359878  0.         -0.52359878 -1.        ]
+result_gt0: [ 1.57079633  0.52359878  0.         -0.5        -1.        ]
+```
```

Comment:
The outputs I got when I executed the code are different from yours mentioned here. Please check those, and I suppose there is some issue with the code as well; because if the where condition is `False`, it is still computing the inverse sine for that element. So, please recheck the code.

Reply:
Please also add a ##Codebyte Example block at the end to showcase the user of the `.arcsin()` method, avoid having an output block for it.

Reply:
Argh... numpy and python doesn't seem to want to do things like I would want them to... Anyway, I've made a more robust example with initializing of result arrays, as it seems that that is the only way to get it to reliably do the expected operation. It also works as a codebyte, but i've added a specific simplified codebyte example as well.

The problem with where seem to be, that numpy seem to cache it's results, which means, that the where clause is actually working, but are not overwriting the previous result values, thus resulting in the output not being as expected if you run it multiple times on the same input. I don't intend to go deeper into debugging that snake nest right now.